src/nupic/research/connections.py

```diff
@@ -73,6 +72,17 @@ def __init__(self,
     # Index of the next synapse to be created
     self._nextSynapseIdx = 0
 
+  def segmentCmp(self, a, b):
+    """
+    A simple comparison function for segments to sort them
+    by their corresponding presynapticCell.
```

Comment:
Segments don't have presynaptic cells. They're on a cell, and they synapse with a bunch of presynaptic cells.